vulnerability-report.md

```diff
@@ -0,0 +1,114 @@
+- By plugging in a random route we were able to induce a 404 error.   That 404 error listed out ever single possible route that is in the system.  This is due tot he fact that Debug is set to True.   The url we plugged in to get this view is:
+    - http://127.0.0.1:8000/taskManager/whatever
+
+
+## What you were able to do with the vulnerability (i.e. what was exposed?)
+- We were able to see every possible url to use and route to.  This can leave your system open and vulnerable with out having the person have to guess url routes one at a time. 
+
+
+## The code that fixes the vulnerability
+- Update the settings.py to:
+    - DEBUG = False
+    - TEMPLATE_DEBUG = False
+
+
+# A6 - Sensitive Data Exposure
+## How you found and exploited the vulnerability, including the code/method you used
+- The hashed information was stored using the weak MD5 hasher.  This was used to be quick and thereofre it takes kacers a short amount of time to break the codes.  To fix this we used:
+    - django.contrib.auth.hashers.PBKDF2SHA1PasswordHasher for PASSWORD_HASHERS in the settings.py
+
+
+## What you were able to do with the vulnerability (i.e. what was exposed?)
+- We could not presonally do anything, but hackers would be able to unhash the exposed hashed passwords very quickly.  
+
+## The code that fixes the vulnerability
+- PASSWORD_HASHERS = ['django.contrib.auth.hashers.PBKDF2SHA1PasswordHasher']
+
+
+
+# A7 - Missing Function Level Access Control
+## How you found and exploited the vulnerability, including the code/method you used
+- We found the manage groups view was open if you were simply authenticated.  This was seen by tring to just change things randomly when not an admin. 
+
+
+## What you were able to do with the vulnerability (i.e. what was exposed?)
+- We were able to change the groups without permission to do so.
+
+## The code that fixes the vulnerability
+- This was fixed by in combination of if user.is_authenticated() and:
+    - user.has_perm('can_change_group')
+
+
+# A8 - Cross-site Request Forgery (CSRF)
+## How you found and exploited the vulnerability, including the code/method you used
+- We went to the the profile form and inspected the form for the user, 
+    - http://localhost:8000/taskManager/profile/
+- From here we were able to copy the form into a new html file and render it with the open logged in window.  We were then able to change the values and apply them to the action of the form that connects to the user ID of the person you viewed:
+    - action="http://localhost:8000/taskManager/profile/2"
+
+    - /Users/jackbot/codefellows/401/django.nV/taskManager/edit_profile.html
+
+
+## What you were able to do with the vulnerability (i.e. what was exposed?)
+- We are able to hijack someone's account and change the information that they had so we can access their studd.
+
+## The code that fixes the vulnerability
+- Remove the @csrf_exempt decorators so that it requires csrf.
```

Comment:
You need a little more than just that. @csrf_exempt will take off the CSRF exemption, sure. But Django's forms don't have CSRF by default. You will have to enable the csrf token in the template